src/nvim/mapping.c

```diff
@@ -1551,6 +1553,16 @@ char_u *eval_map_expr(mapblock_T *mp, int c)
       semsg_multiline("E5108: %s", err.msg);
       api_clear_error(&err);
     }
+    if (p && mp->m_replace_keycodes) {
+      char *buf = NULL;
+      replace_termcodes((char *)p, STRLEN(p), &buf,
```

Comment:
When `replace_termcodes()` is called, do not call `vim_strsave_escape_ks()` below.